src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -384,44 +363,29 @@ .maxstack  4
   IL_0056:  stloc.s    V_4
   IL_0058:  ldloca.s   V_4
```

Comment:
>ldloca.s   V_4 [](start = 12, length = 14)

Is this temp here to avoid bad behavior from repeated reads of the receiver? Just trying to picture what scenarios would cause trouble if the original were used instead of a temp.

Reply:
Right, if the receiver is a property or method call (something with side effects) we don't want to evaluate it multiple times, so we create a temp.